### PR TITLE
Ms2018: Refactor hasOcr()

### PR DESCRIPTION
**Pull request by MingtaoSun**
@scoen @AaronCMuller @snezanam @shuangzhou 
http://ourweb.nla.gov.au/apps/jira/browse/DSCS-2018
- Write a new query to retrieve children by their bib level and sub type

Related to https://github.com/nla/banjo/pull/1900

### CODE REVIEW

File: src/amberdb/query/WorkChildrenQuery.java [hunk 1]
```diff
@@ -316,4 +320,46 @@ public Integer getTotalChildCount(Long workId) {
...
+
+    private Query getChildrenIdsByBibLevelSubTypeQuery(Handle h, List<Long> parentIds, BibLevel bibLevel, SubType subType) {
+        if (subType != null){
+            return h.createQuery(
+                    "select distinct v.id from property p1, property p2, edge e, vertex v " +
+                            "where p1.txn_end = 0 and e.txn_end = 0 and v.txn_end = 0 and p2.txn_end = 0 " +
+                            "and p1.id = e.v_out and p2.id = e.v_out and v.id = p1.id and v.id = p2.id " +
+                            "and e.label = 'isPartOf' " +
+                            "and p1.name = 'bibLevel' and p1.value = :bibLevel " +
+                            "and p2.name = 'subType' and p2.value = :subType" +
```
Comment (shuangzhou) [inline src/amberdb/query/WorkChildrenQuery.java]: @MingtaoSun might be good to add a space after :subType.

File: src/amberdb/query/WorkChildrenQuery.java [hunk 2]
```diff
@@ -316,4 +320,46 @@ public Integer getTotalChildCount(Long workId) {
...
+                            "and e.v_in in ("+ Joiner.on(",").join(parentIds)+")")
+                    .bind("bibLevel", bibLevel.code())
+                    .bind("subType", subType.code())
+                    .map(LongMapper.FIRST);
+        }
+        return h.createQuery(
+                "select distinct v.id from property p1, edge e, vertex v " +
```
Comment (shuangzhou) [inline src/amberdb/query/WorkChildrenQuery.java]: @MingtaoSun are there ever a case that input bibLevel can be null?

File: src/amberdb/query/WorkChildrenQuery.java [hunk 3]
```diff
@@ -316,4 +320,46 @@ public Integer getTotalChildCount(Long workId) {
...
+
+    private Query getChildrenIdsByBibLevelSubTypeQuery(Handle h, List<Long> parentIds, BibLevel bibLevel, SubType subType) {
+        if (subType != null){
+            return h.createQuery(
+                    "select distinct v.id from property p1, property p2, edge e, vertex v " +
+                            "where p1.txn_end = 0 and e.txn_end = 0 and v.txn_end = 0 and p2.txn_end = 0 " +
+                            "and p1.id = e.v_out and p2.id = e.v_out and v.id = p1.id and v.id = p2.id " +
+                            "and e.label = 'isPartOf' " +
+                            "and p1.name = 'bibLevel' and p1.value = :bibLevel " +
+                            "and p2.name = 'subType' and p2.value = :subType " +
```
Comment (MingtaoSun) [inline src/amberdb/query/WorkChildrenQuery.java]: @shuangzhou , space added

File: src/amberdb/query/WorkChildrenQuery.java (hunk 2 above)
Comment (MingtaoSun) [inline src/amberdb/query/WorkChildrenQuery.java]: @shuangzhou , no, at least not when determining whether a work has ocr or not. Bib Level is always required.

**Pull request merged**